### PR TITLE
Add some docs to StatefulBuilder

### PR DESCRIPTION
**Pull request by xster**
StatefulBuilder is strangely spartan. Add some docs.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/basic.dart [hunk 1]
```diff
@@ -6240,6 +6240,45 @@ typedef StatefulWidgetBuilder = Widget Function(BuildContext context, StateSette
...
 ///
+/// The [StateSetter] function passed to the [builder] is used to invoke a
+/// rebuild instead of a typical [State]'s [State.setState].
+///
+/// Since the [builder] is re-invoked when the [StateSetter] is called, any
+/// variables that represents state should be kept outside the [builder] function.
+///
+/// {@tool sample}
+///
+/// This example shows using an inline StatefulBuilder that rebuilds and that
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/basic.dart]: [StatefulBuilder]?

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
Cirrus is not happy though.

**Comment by dnfield** [comment]
@xster will you be continuing with this PR?

**Comment by xster** [comment]
oh interesting. Funny that 

```dart
showDialog();
```

made it assume I was trying to define a function showDialog and 

```dart
showDialog<void>();
```

made it realize it was just an expression.

**Pull request merged**